crates/dojo-world/src/manifest.rs

```diff
@@ -517,12 +517,18 @@ fn parse_models_events(events: Vec<EmittedEvent>) -> Vec<Manifest<DojoModel>> {
     let mut models: HashMap<String, FieldElement> = HashMap::with_capacity(events.len());
 
     for e in events {
-        let model_event = if let WorldEvent::ModelRegistered(m) =
-            e.try_into().expect("ModelRegistered event is expected to be parseable")
-        {
-            m
+        let model_event = if let Ok(m) = e.try_into() {
+            if let WorldEvent::ModelRegistered(mr) = m {
+                mr
+            } else {
+                // should never happen as `events` have already been filtered
+                panic!("ModelRegistered expected");
+            }
         } else {
-            panic!("ModelRegistered expected");
+            // As models were registered with the new event type, we can
+            // skip old ones. We are sure at least 1 new event was emitted
+            // when models were migrated.
+            continue;
```

Comment:
We could re-write this in a more idiomatic way to avoid several branching:
```rust
let model_event = match e.try_into() {
    Ok(WorldEvent::ModelRegistered(mr)) => mr,
    Ok(_) => panic!("ModelRegistered expected, but got a different WorldEvent"),
    Err(_) => {
        // As models were registered with the new event type, we can
        // skip old ones. We are sure at least 1 new event was emitted
        // when models were migrated.
        continue;
    }
};
```

Reply:
Oh nice ! Thanks 👍